packages/inventory/src/containers/InventoryAddEdit/index.tsx

```diff
@@ -0,0 +1,267 @@
+  products: [],
+  fetchProductsCreator: fetchProducts,
+  fetchLocationUnitsCreator: fetchLocationUnits,
+  fetchInventoriesCreator: fetchInventories,
+  servicePoint: null,
+};
+
+const InventoryAddEdit: React.FC<InventoryAddEditProps> = (props: InventoryAddEditProps) => {
+  const {
+    openSRPBaseURL,
+    cancelURL,
+    products,
+    fetchProductsCreator,
+    fetchLocationUnitsCreator,
+    fetchInventoriesCreator,
+    redirectURL,
+    servicePoint,
+    inventory,
+    match,
+    customFetchOptions,
+  } = props;
+  const [UNICEFSections, setUNICEFSections] = React.useState<Setting[]>([]);
+  const [donors, setDonors] = React.useState<Setting[]>([]);
+  const { Title } = Typography;
+  const isEdit = !!match.params[ROUTE_PARAM_INVENTORY_ITEM_ID];
+
+  useEffect(() => {
+    // Handle when servicePoint is null e.g when a user refreshes page
+    if (!servicePoint) {
+      // fetch location units
+      const service = new OpenSRPService(
+        LOCATIONS_GET_ALL_SYNC_ENDPOINT,
+        openSRPBaseURL,
+        customFetchOptions
+      );
+      service
+        .list({
+          serverVersion: 0,
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          is_jurisdiction: false,
+        })
+        .then((response: LocationUnit[]) => {
+          fetchLocationUnitsCreator(response);
+        })
+        .catch((_: HTTPError) => {
+          sendErrorNotification(ERROR_GENERIC);
+        });
+    }
+  }, [servicePoint, fetchLocationUnitsCreator, openSRPBaseURL, customFetchOptions]);
+
+  useEffect(() => {
+    if (!products.length) {
+      loadProductCatalogue(
+        openSRPBaseURL,
+        OpenSRPService,
+        fetchProductsCreator
+      ).catch((_: HTTPError) => sendErrorNotification(ERROR_GENERIC));
+    }
+  }, [products.length, openSRPBaseURL, fetchProductsCreator]);
+
+  useEffect(() => {
+    fetchSettings(openSRPBaseURL, { serverVersion: 0, identifier: 'inventory_donors' }, setDonors);
+    fetchSettings(
+      openSRPBaseURL,
+      { serverVersion: 0, identifier: 'inventory_unicef_sections' },
+      setUNICEFSections
+    ); // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    // Handle when editing an item and inventory is null e.g when user refreshes page
+    if (isEdit && !inventory && servicePoint) {
+      // fetch inventories
+      const service = new OpenSRPService(
+        `${OPENSRP_ENDPOINT_GET_INVENTORIES}/${servicePoint.id}`,
+        openSRPBaseURL,
+        customFetchOptions
+      );
+      service
+        .list()
+        .then((response: Inventory[]) => {
+          fetchInventoriesCreator(response);
+        })
+        .catch((_: HTTPError) => {
+          sendErrorNotification(ERROR_GENERIC);
+        });
+    }
+  }, [
+    customFetchOptions,
+    isEdit,
+    inventory,
+    servicePoint,
+    openSRPBaseURL,
+    fetchInventoriesCreator,
+  ]);
+
+  if (!servicePoint) {
+    return <Spin size="large" />;
+  }
```

Comment:
Shouldn't we show some of message that the page was not able to render due to this, as opposed to showing an eternal loader

Reply:
A `servicePoint` will be null if the service point was not present in the store, and in that case, a fetch will be made. In case this fails, it will display the error notification message

Reply:
we have a Resouce404 component we can use to show a certain resource was not found: `reactutils.Resrouce404`

Reply:
@p-netm Ok..we can not use a 404 page here because basically the loader is displayed before the service point becomes available...Think of it like how we make use of the spinner in other components

Reply:
correct me if am wrong; we the use the spinner to represent loading states, loading state is then based on whether some important promise(s) were resolved regardless of whether the promises resolved with some truthful data.

Reply:
Yea sure. I would say it applies the same here. If a user refreshes the page, the spinner will be shown as we get the service point, because the form cannot be displayed/work if the service point is null

Reply:
yeah and when the promise resolves without a truthy servic epoint, then we return a Resource404 page instead of keeping the spinner in the ui.

Reply:
@p-netm Ok...Not unless we check the status code returned to assert it's a 404...the response can be falsy because of other errors as well at we shouldn't then display a 400 page